com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/transaction/ZigBeeTransactionManager.java

```diff
@@ -639,35 +639,39 @@ private void sendNextTransaction() {
 
             ZigBeeTransaction transaction;
 
-            // Randomly loop through all queues, taking a transaction from each one in turn
+            // Loop through all queues, taking a transaction from each one in turn
             // If we have more transactions outstanding than we're allowed, then exit
             // If we get through an iteration of all queues without sending anything, then exit
             //
             // Points to note -:
             // * Queues may have more than one transaction to send
             // * Queues may have transactions to send, but be unable to send them at this time
 
-            List<ZigBeeTransactionQueue> emptyQueues = new ArrayList<>();
-            boolean sendDone;
+            List<ZigBeeTransactionQueue> sleepyOrDelayedQueues = new ArrayList<>();
+
+            int transactionsSent;
             do {
                 // Exit unless we send at least one transaction
-                sendDone = true;
+                transactionsSent = 0;
```

Comment:
Why change this from a `boolean` to an `int`? You still only check if it is >0, so a boolean would have been fine?

Reply:
Counting seemed more natural. It has been changed back minimize changes.